### PR TITLE
[WIP] Nightwatch tests

### PR DESCRIPTION
**Pull request by joannecheng**
## What does this PR do? How does it affect users?

This starts to add [nightwatch](http://nightwatchjs.org) tests to keen-dataviz. Currently this just tests against the demo page (`test/demo/index.html`)

## How should this be tested?

* Install selenium server (`brew install selenium-server-standalone`)
* Run selenium server (`brew info selenium-server-standalone`)
* Run local project (`gulp`)
* Run `npm install`
* Run `nightwatch`
* Watch tests pass!

Step through the code line by line. Things to keep in mind as you review:
 - Are there any edge cases not covered by this code?
 - Does this code follow conventions (naming, formatting, modularization, etc) where applicable?

Fetch the branch and/or deploy to staging to test the following:

- [x] Does the code compile without warnings (check shell, console)?
- [x] Do all tests pass?
- [x] Does the UI, pixel by pixel, look exactly as expected (check various screen sizes, including mobile)?
- [ ] If the feature makes requests from the browser, inspect them in the Web Inspector. Do they look as expected (parameters, headers, etc)?
- [ ] If the feature sends data to Keen, is the data visible in the project if you run an extraction (include link to collection/query)?
- [ ] If the feature saves data to a database, can you confirm the data is indeed created in the database?

### CODE REVIEW

**Comment by sfalkoff** [comment]
@joannecheng I had to install nightwatch globally to kick off the tests with `nightwatch`. Is there another test runner set up that I may be missing?

**Comment by joannecheng** [comment]
@sfalkoff I believe you need to install nightwatch globally

**Comment by sfalkoff** [comment]
Great, thanks. How about adding these test setup/instructions to the readme?

File: test/nightwatch/donut/donut-test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        [ 'A', 189.98 ],
+        [ 'B', 135.83 ],
+        [ 'C', 124.56 ],
+        [ 'D', 117.22 ],
+        [ 'E', 55.12  ]
+      ];
+
+      window.donut = new Dataviz()
+        .el('.chart-c3-donut')
+        .title('Mmmm, donuts...')
```
Comment (sfalkoff) [inline test/nightwatch/donut/donut-test.js]: nice titles (_she says as she looks up the nearest place to buy donuts_) 😄

File: test/nightwatch/gauge/gauge-test.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    // Notes test
+    browser.expect.element('.chart-c3-gauge .custom-theme-notes')
+      .text.to.equal('Sample notes go down here');
+
+    // SVG Element presence test
+    browser.expect.element('.chart-c3-gauge .c3 svg')
+      .to.be.present.before(1000);
+
+    // Value test - is this necessary?
+    browser.expect.element('.chart-c3-gauge .c3-target-Percentage-of-awesome text')
```
Comment (sfalkoff) [inline test/nightwatch/gauge/gauge-test.js]: this seems like a reasonable thing to add to me

File: test/nightwatch/table/table-test.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+  '@tags': ['custom', 'table'],
+  'disabled': true,
```
Comment (sfalkoff) [inline test/nightwatch/table/table-test.js]: just curious - are the table tests disabled because they are a work in progress?

**Comment by sfalkoff** [comment]
@joannecheng I think this looks good. Adding tests for the x-axis labels and tooltips like you mentioned in your TODO comments sounds like a good idea for the next iteration.

File: test/nightwatch/table/table-test.js (hunk 3 above)
Comment (joannecheng) [inline test/nightwatch/table/table-test.js]: Yes, and also because I was trying to figure out how to test each cell in a table. Hopefully will get this sorted out before this PR gets merged.

**Comment by dustinlarimer** [comment]
@joannecheng @sfalkoff I'm happy merging this whenever y'all are 👍

**Comment by sfalkoff** [comment]
👍
